Jingle/SCJingleConverter/composition.cpp

```diff
@@ -849,37 +859,137 @@ uint32_t Composition::getMemUsage() {
  *      configured Measure Start and Stop Indices.
  */
 uint32_t Composition::getNumNotes(Channel chan) {
+    if (chan == RIGHT) {
+        return optimizedNotesR.size();
+    } else {
+        return optimizedNotesL.size();
+    }
+}
+
+/**
+ * @brief Composition::optimizeNotes Condenses and trims notes to take up
+ *  less space when downloaded to the Steam Controller.
+ */
+void Composition::optimizeNotes() {
     const uint32_t meas_start_idx = getMeasStartIdx();
     const uint32_t meas_end_idx = getMeasEndIdx();
 
-    uint32_t num_notes = 0;
+    optimizedNotesL.clear();
+    optimizedNotesR.clear();
 
-    Voice* voice = nullptr;
-    switch (chan) {
-    case RIGHT:
-        if (voices.count(voiceStrR)) {
-            voice = &voices[voiceStrR];
+    uint32_t removed_notes = 0;
+
+    // Optimize Notes in Left Channel
+    if (voices.count(voiceStrL)) {
+        // Collect notes into single vector
+        for (uint32_t meas_idx = meas_start_idx; meas_idx <= meas_end_idx; meas_idx++) {
+            Measure& meas = voices[voiceStrL].measures[meas_idx];
+            for (uint32_t notes_idx = 0; notes_idx < meas.notes.size(); notes_idx++) {
+                optimizedNotesL.push_back(meas.notes[notes_idx]);
+            }
```

Comment:
Assuming meas.notes is a vector, you can replace this loop with:

```cpp
optimizedNotesL = meas.notes
```

Reply:
It is, but I don't think we can. We're appending multiple measures to optimizedNotesL.

Reply:
Can't reply to the comment on composistion.cpp for some reason but that sounds good to me. For some reason thought we were limited to uint32_t or maybe just unsigned ints. In hindsight that would have been a pretty easy thing to change and test lol.

Reply:
Oops, sorry, missed the context! In that case, does std::vector::insert help?

```cpp
optimizedNotesL.insert(optimizedNotesL.end(), meas.notes.begin(), meas.notes.end());
```

Reply:
Looks pretty great to me, yeah!